diff --git a/xmpp/transports.py b/xmpp/transports.py
index ed56785..6f508fc 100644
--- a/xmpp/transports.py
+++ b/xmpp/transports.py
@@ -384,7 +384,11 @@ def _startSSL(self):
         """ Immidiatedly switch socket to TLS mode. Used internally."""
         """ Here we should switch pending_data to hint mode."""
         tcpsock=self._owner.Connection
-        tcpsock._sslObj    = ssl.wrap_socket(tcpsock._sock, None, None)
+        context=ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+        context.check_hostname = False
+        context.verify_mode = ssl.CERT_NONE
+
+        tcpsock._sslObj    = context.wrap_socket(tcpsock._sock)
         tcpsock._sslIssuer = tcpsock._sslObj.getpeercert().get('issuer')
         tcpsock._sslServer = tcpsock._sslObj.getpeercert().get('server')
         tcpsock._recv = tcpsock._sslObj.read
@@ -694,7 +698,7 @@ def send(self, raw_data, headers={}):
         default['Content-Length'] = len(bosh_data)
         if self.GZIP:
             default['Accept-Encoding'] = 'gzip, deflate'
-        headers = dict(default, **headers) 
+        headers = dict(default, **headers)
         conn = self.Connection()
         if self.PIPELINE:
             conn._HTTPConnection__state = _CS_IDLE
